include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -413,7 +413,8 @@ class __future : public __future_base
     _T
     get()
     {
-        return __data.template get_access<access_mode::read>()[__result_idx];
+        auto __sub_buffer = sycl::buffer{__data, sycl::id<1>(__result_idx), sycl::range<1>(1)};
```

Comment:
1) Probably, make sense to create a sub-buffer on future instance creating in the SYCL backend pattern? I mean line `return oneapi::dpl::__par_backend_hetero::__future<_Tp>(__reduce_event, __offset_2, __temp);`
2) Probably, we don't need to keep `__result_idx` field... just keep `buffer` (sub-buffer) which  contains just one interesting value where __result_idx=0 ?

Reply:
Actually, there is no "copying" optimization in case of sub-buffer in the current implementation. And I guess it is quite complicated to effective implement. So, I suggest using a separate `sycl::buffer`.